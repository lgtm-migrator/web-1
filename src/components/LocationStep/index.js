import React from "react"
import { Form, H2, Paragraph } from "muy"
import Button from "@material-ui/core/Button"
import Alert from "@material-ui/lab/Alert"
import AlertTitle from "@material-ui/lab/AlertTitle"
import FormControl from "@material-ui/core/FormControl"
import EmbedGoogleMap from "../../containers/EmbedGoogleMap"
import Skeleton from "@material-ui/lab/Skeleton"
import Fade from "@material-ui/core/Fade"
import FormHelperText from "@material-ui/core/FormHelperText"
import Divider from "@material-ui/core/Divider"
import { GoogleMaps } from "../GoogleMaps"
import useFeatureToggle from "../../hooks/useFeatureToggle"

const LocationStep = (props) => {
  const {
    checkingGeolocationSupport,
    children,
    currentPosition,
    error,
    hasGeolocationSupport,
    loadingCurrentLocation,
    loadingMap,
    onGetLocationClick,
    onMapLoaded,
    permissionDenied,
    isLocationValid,
  } = props

  const [googleMapsJsToggle] = useFeatureToggle("GOOGLE_MAPS_JS")

  const getButtonLabel = () => {
    if (checkingGeolocationSupport) {
      return <>Carregando&hellip;</>
    }
    if (permissionDenied) {
      return <>Tentar novamente</>
    }
    if (loadingCurrentLocation) {
      return <>Obtendo localização&hellip;</>
    }
    if (loadingMap) {
      return <>Carregando mapa&hellip;</>
    }
    if (currentPosition) {
      return <>Atualizar localização</>
    }
    return <>Permitir uso da localização</>
  }

  return (
    <Form>
      <H2 gutterBottom={true}>Envie a localização da vaga</H2>

      <FormControl component={"fieldset"} error={error}>
        {permissionDenied && (
          <Alert severity={"error"}>
            <AlertTitle>Não é possível saber a localização da vaga</AlertTitle>
            Por favor, dê permissão para uso da localização e tente novamente.
          </Alert>
        )}
        <Divider />
        <Paragraph align="center">
          <Fade in={hasGeolocationSupport}>
            <Button
              disabled={
                checkingGeolocationSupport ||
                !hasGeolocationSupport ||
                loadingCurrentLocation ||
                loadingMap
              }
              onClick={onGetLocationClick}
              variant={"contained"}
            >
              {getButtonLabel()}
            </Button>
          </Fade>
        </Paragraph>
        <FormHelperText>
          Utilizamos seus dados de localização apenas para ajudar os órgãos a
          encontrarem o local da infração.
        </FormHelperText>
      </FormControl>
      <Divider />
      {currentPosition && (
        <div style={{ position: "relative" }}>
          {loadingMap && (
            <Skeleton
              style={{ position: "absolute", zIndex: 2 }}
              variant="rect"
              width={"100%"}
              height={157}
            />
          )}

          <EmbedGoogleMap
            coordinates={`${currentPosition.coords.latitude},${currentPosition.coords.longitude}`}
            onLoad={onMapLoaded}
            style={{ height: 157, width: "100%" }}
          />

          {googleMapsJsToggle && <GoogleMaps />}

          <Paragraph variant={"body2"}>
            Se estiver tudo certo com esta localização, você pode finalizar a
            denúncia.
          </Paragraph>
        </div>
      )}

      {children}

      {!isLocationValid && (
        <Alert severity={"warning"} variant="outlined">
          <AlertTitle>Algo não deu certo!</AlertTitle>
          Por favor, tente novamente.
        </Alert>
      )}
    </Form>
  )
}

export default LocationStep
