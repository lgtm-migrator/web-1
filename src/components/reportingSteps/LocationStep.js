import React from "react"
import { Figure, Form, H2, Paragraph } from "muy"
import Button from "@material-ui/core/Button"
import Alert from "@material-ui/lab/Alert"
import AlertTitle from "@material-ui/lab/AlertTitle"
import FormControl from "@material-ui/core/FormControl"
import EmbedGoogleMap from "../../containers/EmbedGoogleMap"
import Skeleton from "@material-ui/lab/Skeleton"
import Fade from "@material-ui/core/Fade"
import FormHelperText from "@material-ui/core/FormHelperText"
import Divider from "@material-ui/core/Divider"

const LocationStep = (props) => {
  const {
    checkingGeolocationSupport,
    children,
    currentPosition,
    error,
    hasGeolocationSupport,
    loadingCurrentLocation,
    loadingMap,
    onGetLocationClick,
    onMapLoaded,
    permissionDenied,
  } = props

  const getButtonLabel = () => {
    if (checkingGeolocationSupport) {
      return <>Carregando&hellip;</>
    }
    if (permissionDenied) {
      return <>Tentar novamente</>
    }
    if (loadingCurrentLocation) {
      return <>Obtendo localização&hellip;</>
    }
    if (loadingMap) {
      return <>Carregando mapa&hellip;</>
    }
    if (currentPosition) {
      return <>Atualizar localização</>
    }
    return <>Permitir uso da localização</>
  }

  return (
    <Form>
      <H2 gutterBottom={true}>Envie a localização da vaga</H2>

      <FormControl component={"fieldset"} error={error}>
        {permissionDenied && (
          <Alert severity={"error"}>
            <AlertTitle>Não é possível saber a localização da vaga</AlertTitle>
            Por favor, dê permissão para uso da localização e tente novamente.
          </Alert>
        )}
        <Divider />
        <Paragraph align="center">
          <Fade in={hasGeolocationSupport}>
            <Button
              disabled={
                checkingGeolocationSupport ||
                !hasGeolocationSupport ||
                loadingCurrentLocation ||
                loadingMap
              }
              onClick={onGetLocationClick}
              variant={"contained"}
            >
              {getButtonLabel()}
            </Button>
          </Fade>
        </Paragraph>
        <FormHelperText>
          Utilizamos seus dados de localização apenas para ajudar os órgãos a
          encontrarem o local da infração.
        </FormHelperText>
      </FormControl>
      <Divider />
      {currentPosition && (
        <>
          {loadingMap && (
            <Skeleton variant="rect" width={"100%"} height={157} />
          )}

          <EmbedGoogleMap
            coordinates={`${currentPosition.coords.latitude},${currentPosition.coords.longitude}`}
            onLoad={onMapLoaded}
            style={{ height: 157, width: "100%" }}
          />

          <Paragraph variant={"body2"}>
            Se estiver tudo certo com esta localização, você pode finalizar a
            denúncia.
          </Paragraph>
        </>
      )}

      {children}
    </Form>
  )
}

export default LocationStep
