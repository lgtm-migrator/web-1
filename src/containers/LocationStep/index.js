import React, { useEffect, useState } from "react"
import usePosition from "../../hooks/usePosition"
import LocationStep from "../../components/LocationStep"
import { useSetParkingReportState } from "../../hooks/useParkingReportState"
import { useSetLoadingState } from "../../hooks/useLoadingState"

export default () => {
  const {
    checkingGeolocationSupport,
    currentPosition,
    error,
    hasGeolocationSupport,
    getCurrentPosition,
    loadingCurrentLocation,
    permissionDenied,
  } = usePosition()

  const setLoading = useSetLoadingState()
  const [loadingMap, setLoadingMap] = useState(null)
  const setParkingReportState = useSetParkingReportState()

  useEffect(() => {
    if (Boolean(currentPosition)) {
      setLoadingMap(true)
      setLoading(true)
    }
  }, [currentPosition, setLoading])

  useEffect(() => {
    setParkingReportState((oldParkingReportState) => ({
      ...oldParkingReportState,
      currentPosition,
    }))
  }, [currentPosition, setParkingReportState])

  const handleMapLoaded = () => {
    setLoadingMap(false)
    setLoading(false)
  }

  const handleGetLocationClick = () => {
    getCurrentPosition()
    setLoadingMap(true)
    setLoading(true)
  }

  return (
    <LocationStep
      checkingGeolocationSupport={checkingGeolocationSupport}
      currentPosition={currentPosition}
      error={error}
      hasGeolocationSupport={hasGeolocationSupport}
      loadingCurrentLocation={loadingCurrentLocation}
      loadingMap={loadingMap}
      onGetLocationClick={handleGetLocationClick}
      onMapLoaded={handleMapLoaded}
      permissionDenied={permissionDenied}
    />
  )
}
