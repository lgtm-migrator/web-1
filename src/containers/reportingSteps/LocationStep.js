import React, { useEffect, useState } from "react"
import usePosition from "../../hooks/usePosition"
import LocationStep from "../../components/reportingSteps/LocationStep"
import useGlobal from "../../hooks/useGlobal"

export default () => {
  const {
    checkingGeolocationSupport,
    currentPosition,
    error,
    hasGeolocationSupport,
    getCurrentPosition,
    loadingCurrentLocation,
    permissionDenied,
  } = usePosition()

  const [globalState, globalActions] = useGlobal()
  const [loadingMap, setLoadingMap] = useState(null)

  useEffect(() => {
    if (currentPosition) {
      setLoadingMap(true)
      globalActions.setCurrentParkingReportingData({
        ...globalState.currentParkingReportingData,
        currentPosition,
      })
    }
  }, [currentPosition])

  const handleMapLoaded = () => {
    setLoadingMap(false)
  }

  const handleGetLocationClick = () => {
    getCurrentPosition()
    setLoadingMap(true)
  }

  return (
    <LocationStep
      checkingGeolocationSupport={checkingGeolocationSupport}
      currentPosition={currentPosition}
      error={error}
      hasGeolocationSupport={hasGeolocationSupport}
      loadingCurrentLocation={loadingCurrentLocation}
      loadingMap={loadingMap}
      onGetLocationClick={handleGetLocationClick}
      onMapLoaded={handleMapLoaded}
      permissionDenied={permissionDenied}
    />
  )
}
