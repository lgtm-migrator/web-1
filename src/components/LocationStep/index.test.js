import React from "react"
import renderer from "react-test-renderer"
import { render, screen } from "@testing-library/react"
import LocationStep from "."

describe("LocationStep", () => {
  let props

  beforeEach(() => {
    props = {
      permissionDenied: false,
      isLocationValid: true,
    }
  })

  it("should render correctly", () => {
    const tree = renderer.create(<LocationStep {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render warning panel", () => {
    props.isLocationValid = false
    render(<LocationStep {...props} />)

    const element = screen.getByText("Algo não deu certo!")
    expect(element).toBeTruthy()
  })

  it("should render error panel", () => {
    props.permissionDenied = true
    render(<LocationStep {...props} />)

    const element = screen.getByText(
      "Não é possível saber a localização da vaga"
    )
    expect(element).toBeTruthy()
  })
})
