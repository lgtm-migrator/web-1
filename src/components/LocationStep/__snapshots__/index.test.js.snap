// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationStep should render correctly 1`] = `
<form
  className="MuiBox-root MuiBox-root-2"
  data-name="form"
  id="form"
  name="form"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
  >
    Envie a localização da vaga
  </h2>
  <fieldset
    className="MuiFormControl-root"
  >
    <hr
      className="MuiDivider-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph MuiTypography-alignCenter"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "opacity": 0,
            "visibility": "hidden",
          }
        }
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Permitir uso da localização
        </span>
      </button>
    </p>
    <p
      className="MuiFormHelperText-root"
    >
      Utilizamos seus dados de localização apenas para ajudar os órgãos a encontrarem o local da infração.
    </p>
  </fieldset>
  <hr
    className="MuiDivider-root"
  />
</form>
`;
